src/talk/renderer/TalkWrapper/TalkWrapper.vue

```diff
@@ -4,38 +4,85 @@
   -->
 
 <script setup lang="ts">
-import { onMounted } from 'vue'
+import { onMounted, onUnmounted } from 'vue'
 import { onTalkHashDirty, onTalkHashUpdate, openConversation, setTalkHash } from './talk.service.ts'
 import { registerTalkDesktopSettingsSection } from '../Settings/index.ts'
 import { subscribeBroadcast } from '../../../shared/broadcast.service.ts'
 import { appData } from '../../../app/AppData.js'
+import unreadCountStore, { EVENTS } from '@talk/src/store/unreadCountStore'
+import { subscribe, emit as emitEvent } from '@nextcloud/event-bus'
 
 const emit = defineEmits<{
 	(event: 'ready'): void
 }>()
 
+// Handler for unread count updates
+let unsubscribeUnreadCount: (() => void) | null = null
+
+const waitForTalkAndStore = async () => {
+	console.debug('TalkWrapper: Waiting for Talk instance and store...')
+	let attempts = 0
+	const maxAttempts = 50
+	const delay = 100
+
+	while (attempts < maxAttempts) {
+		if (window.OCA?.Talk?.instance && window.store && window.TALK_DESKTOP) {
+			console.debug('TalkWrapper: Talk instance, store, and TALK_DESKTOP found')
+			return true
+		}
+		await new Promise(resolve => setTimeout(resolve, delay))
+		attempts++
+	}
+	throw new Error('Talk instance, store, or TALK_DESKTOP not initialized after maximum attempts')
+}
+
 onMounted(async () => {
-	// Importing the main Talk entry point mounts a Vue app to the #content
-	await import('@talk/src/main.js')
+	try {
+		await import('@talk/src/main.js')
+		
+		registerTalkDesktopSettingsSection()
+		subscribeBroadcast('talk:conversation:open', ({ token, directCall }) => openConversation(token, { directCall }))
+
+		await waitForTalkAndStore()
+		const store = window.store
+
+		// Register store module if not already registered
+		if (!store.hasModule('unreadCount')) {
+			store.registerModule('unreadCount', unreadCountStore)
+		}
 
-	// Additional integrations
-	registerTalkDesktopSettingsSection()
-	subscribeBroadcast('talk:conversation:open', ({ token, directCall }) => openConversation(token, { directCall }))
+		// Initialize badge handling
+		window.TALK_DESKTOP.setBadgeCount(0)
+
+		// Subscribe to unread count updates
+		unsubscribeUnreadCount = subscribe(EVENTS.UNREAD_COUNT_UPDATED, ({ unreadMessages }) => {
+			window.TALK_DESKTOP.setBadgeCount(unreadMessages || 0)
+		})
```

Comment:
IMO this is the only part that belongs to Desktop client:
- it should be subscribed to talk event
- it should update counter based on internal logic and client config (say, user wants to ignore them at all and don't show a badge / or count only unread mentions / or all of unread messages)

Store should be properly registered in the app itself

Reply:
Not that I want anyone to rush things, but I understood correctly that we shall wait for more input from your devs?

Reply:
@ShGKme wanted to post feedback soon